src/backend/joanie/core/models/products.py

```diff
@@ -397,6 +426,49 @@ def can_edit(self):
         """Return True if the order group can be edited."""
         return not self.orders.exists()
 
+    @property
+    def is_enabled(self):
+        """
+        Returns boolean whether the order group is active based on its status, availability,
+        and time constraints.
+        When a start or end date is setted, we need to verify if we are in the period
+        of early bird, or last minute depending on the course run enrollment start or
+        enrollment end datetimes. When both dates are setted, we need to make sure that the
+        current day is between those.
+        """
+        if not self.is_active:
+            return False
+
+        # Unlimited spots
+        if self.is_active and self.nb_seats is None:
+            return True
+
+        available_seats = self.nb_seats - self.get_nb_binding_orders()
+        if not available_seats:
+            return False
+
+        if self.course_product_relation.product.type == enums.PRODUCT_TYPE_CERTIFICATE:  # pylint: disable=no-member
+            instance = self.course_product_relation.course
+        else:
+            instance = self.course_product_relation.product
+        course_run_dates = instance.get_equivalent_course_run_dates(  # pylint: disable=no-member
+            ignore_archived=True
+        )
+
+        is_active = False
+        now = timezone.now()
+
+        if not self.start and not self.end:
+            is_active = self.is_active
+        elif self.start and self.end:
+            is_active = self.start <= now <= self.end
+        elif self.end:  # Early birds
+            is_active = self.end >= now >= course_run_dates["enrollment_start"]
+        elif self.start:  # Last minutes
+            is_active = self.start <= now <= course_run_dates["enrollment_end"]
+
+        return is_active
```

Comment:
```suggestion
        if not self.is_active:
            return False

        # Unlimited spots
        if self.nb_seats is None:
            return True

        if self.nb_seats - self.get_nb_binding_orders() == 0:
            return False

        if not self.start and not self.end:
            return True

        now = timezone.now()
        if self.start and self.end:
            return self.start <= now <= self.end
        
        if self.course_product_relation.product.type == enums.PRODUCT_TYPE_CERTIFICATE:
            course_run_dates = self.course_product_relation.course.get_equivalent_course_run_dates(
                ignore_archived=True
            )
        else:
            course_run_dates = self.course_product_relation.product.get_equivalent_course_run_dates(
                ignore_archived=True
            )

        if self.end:  # Early birds
            return course_run_dates["enrollment_start"] <= now <= self.end
        elif self.start:  # Last minutes
            return self.start <= now <= course_run_dates["enrollment_end"]

        return False
```
Sorry for all this rewrite, but I find is simpler like this. WDYT ?

Reply:
No worries ! Thanks for the rewrite :pray:  ! It's actually much simpler to understand at first sight.